### ISSUE TITLE
Could not resolve placeholder 'xxxx' in string value "${yyyy}" when used placeholder resolver in spring configurations

### ISSUE THREAD

**Issue by sabyrzhan** [springfox/springfox#677]
java.lang.IllegalArgumentException: Could not resolve placeholder 'xxxx' in string value "${yyyy}" when using configuration placeholder in spring based project (at least in spring xml since I am using spring integration in my project). The exception started to happen when i added springfox-swagger2 dependency to maven project. After digging into the problem i found out that there might be a problem with PropertySourcesPlaceholderConfigurer and set: setIgnoreUnresolvablePlaceholders(true) the error disappeared. The modified code in SwaggerCommonConfiguration.java from springfox-swagger-common for my solution is below:
```java
@Bean
  public static PropertySourcesPlaceholderConfigurer swaggerProperties() {
        PropertySourcesPlaceholderConfigurer p = new PropertySourcesPlaceholderConfigurer();
        p.setIgnoreUnresolvablePlaceholders(true);
        return p;
  }
```

Tools i used:
1) java version "1.8.0_31"
Java(TM) SE Runtime Environment (build 1.8.0_31-b13)
Java HotSpot(TM) 64-Bit Server VM (build 25.31-b07, mixed mode)
2) Spring Integration 4.0.6.RELEASE
3) Spring Context 4.0.6.RELEASE
4) Spring Boot 1.1.5.RELEASE
5) Apache Maven 3.0.5
6) OS Linux Mint 17.1 64-bit
7) Springfox Swagger 2.0.0-SNAPSHOT

**Comment by adrianbk** [comment]
@sabyrzhan good catch! This need to be fixed before 2.0 is released.

**Comment by adrianbk** [comment]
@sabyrzhan trying to find out what exact condition causes this - have you any special propertyplaceholder beans or configuration?

**Comment by sabyrzhan** [comment]
So how is it in my project: 
1) I used <context:property-placeholder location="file://${property_location}" /> in my spring-integration XML file. 
2) XML file locate in classpath: /META-INF/integration-context.xml
3) And imported that file in my configuration bean:

```java
@Configuration
@ImportResource("classpath:/META-INF/integration-context.xml")
public class IntegrationConfiguration {
    private static final Logger LOGGER = LoggerFactory.getLogger(IntegrationConfiguration.class);
.
.
.
```

**Comment by dilipkrish** [comment]
@adrianbk It's not clear to me why this a problem. This change will cause property resolution to fail silently. Is that what we want?

**Comment by sabyrzhan** [comment]
@dilipkrish Does not this let other resolvers to do property resolution if $this resolver cannot do that?

**Comment by adrianbk** [comment]
@dilipkrish as far as I am aware spring uses all PropertySourcesPlaceholderConfigurer's (PPC) in a cain to resolve a given property. If the springfox PPC happens to be used first to resolve some non-springfox property then, without this change,  an Exception is thrown.

**Comment by adrianbk** [comment]
Because the placeholder properties we are using have defaults:`"${springfox.documentation.swagger.v1.path:/api-docs}"` we don't break other PropertySourcesPlaceholderConfigurer which don't have`ignoreUnresolvablePlaceholders` set to true.

**Comment by dilipkrish** [comment]
@adrianbk @sabyrzhan leaving unresolved variables seems wrong to me some how. What do you guys think about scoping the resolution via a prefix and suffix.

something like ```$springfox{springfox.documentation.swagger.v1.path:/api-docs}``` that way its scoped and ignoring will not affect downstream libraries? scoping = it will only try to resolve properties that begin with **$springfox**.

**Comment by adrianbk** [comment]
not sure the end result would be much different. is there any way around this which doesn't involve using a PPC?

**Comment by dilipkrish** [comment]
Unfortunately not :( let me add the scoping prefix. It will at the very least not try and resolve every variable. Springboot has a nice way around it but that won't work with vanilla spring apps

**Comment by sanndeepg** [comment]
Configure PropertySourcesPlaceholderConfigurer in the container or bean

Refer More for easy configuration
https://in.techiehints.com/wps/2017/12/17/property-file-configuration-for-java-bean-in-spring/

**Comment by dilipkrish** [comment]
Thanks @sandeepg. This is an old issue however and this problem is solved in 2.7.0

**Comment by sanndeepg** [comment]
Thanks, how ever i saw more questions discussion in other sites as well
thought of sharing :)

Thanks

Sandeep G
